jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -587,10 +586,7 @@ ecma_builtin_array_prototype_object_push (ecma_value_t this_arg, /**< this argum
                     ecma_builtin_array_prototype_helper_set_length (obj_p, n),
                     ret_value);
 
-    ecma_number_t *ret_num_p = ecma_alloc_number ();
-    *ret_num_p = n;
-
-    ret_value = ecma_make_number_value (ret_num_p);
+    ret_value = ecma_make_number_value (n);
```

Comment:
Array length range was very confusing for me. I had the impression that it can go up to UINT32_MAX+1, but the spec says otherwise:

http://www.ecma-international.org/ecma-262/5.1/#sec-15.4

Every Array object has a length property whose value is always a nonnegative integer less than 2^32.

This is not the only place where it is sub-optimal, I think we could improve a lot of code then.